### PR TITLE
Tools: Use material symbols in UIs

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Add material symbols to existing tools in ayon-core.

## Additional info
Added `qtmaterialsymbols` custom implementation (similar to `qtawesome`) and use it in basic tools.

### Screenshots
image

## Testing notes:
1. Tools should show same icons as AYON web UI does

### CODE REVIEW

**Comment by BigRoy** [comment]
Starting to look nice! :)

image

No icons in scene inventory yet though - but hey, this is still just a draft PR

image

**Comment by iLLiCiTiT** [comment]
> What's needed for this PR to be ready for review?

Simple folders widget don't use icons (should be simple modification of `SimpleFoldersController`). And select version combobox in scene inventory does not paint the icon.

File: client/ayon_core/tools/utils/delegates.py [hunk 1]
```diff
@@ -130,44 +130,107 @@ def paint(self, painter, option, index):
...
 
         painter.drawText(
-            padded_text_rect,
+            text_rect,
             option.displayAlignment,
             text
         )
 
+        if option.state & QtWidgets.QStyle.State_HasFocus:
+            focus_opt = QtWidgets.QStyleOptionFocusRect(option)
```
Comment (BigRoy) [inline client/ayon_core/tools/utils/delegates.py:205]: Getting this in Fusion:
```
Traceback (most recent call last):
  File "E:\dev\ayon-core\client\ayon_core\tools\utils\delegates.py", line 205, in paint
    focus_opt = QtWidgets.QStyleOptionFocusRect(option)
TypeError: 'PySide2.QtWidgets.QStyleOptionFocusRect' called with wrong argument types:
  PySide2.QtWidgets.QStyleOptionFocusRect(QStyleOptionViewItem)
Supported signatures:
  PySide2.QtWidgets.QStyleOptionFocusRect()
  PySide2.QtWidgets.QStyleOptionFocusRect(PySide2.QtWidgets.QStyleOptionFocusRect)
  PySide2.QtWidgets.QStyleOptionFocusRect(int)
```
From: https://github.com/ynput/ayon-core/pull/593

File: client/ayon_core/tools/utils/lib.py [hunk 2]
```diff
@@ -468,7 +469,7 @@ def _get_cache_key(cls, icon_def):
         if icon_type == "path":
             parts = [icon_type, icon_def["path"]]
 
-        elif icon_type == "awesome-font":
+        elif icon_type in {"awesome-font", "material-symbols"}:
             parts = [icon_type, icon_def["name"], icon_def["color"]]
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply client/ayon_core/tools/utils/lib.py:473]: Fixed by not passing in `QColor` object.

[the remark this answers is not in the document]
Comment (BigRoy) [reply client/ayon_core/tools/utils/lib.py:473]: Even better. 👍

**Comment by BigRoy** [comment]
[This](https://github.com/ynput/ayon-core/pull/593#discussion_r1633190232) is now fixed with https://github.com/ynput/ayon-core/pull/593/commits/909b6b9d1f244d34effd0cc219dd57663bb96b0e

But I'm now getting:
```python
Traceback (most recent call last):
  File "E:\dev\ayon-core\client\ayon_core\tools\utils\delegates.py", line 223, in paint
    QtWidgets.QPalette.Normal
AttributeError: module 'qtpy.QtWidgets' has no attribute 'QPalette'
```

File: client/ayon_core/tools/utils/delegates.py [hunk 3]
```diff
@@ -130,44 +130,113 @@ def paint(self, painter, option, index):
...
+                    QtWidgets.QPalette.Normal
+                    if option.state & QtWidgets.QStyle.State_Enabled
+                    else QtWidgets.QPalette.Disabled
+                ),
+                (
+                    QtWidgets.QPalette.Highlight
+                    if option.state & QtWidgets.QStyle.State_Selected
+                    else QtWidgets.QPalette.Window
+                )
+            )
```
Comment (BigRoy) [inline client/ayon_core/tools/utils/delegates.py:232]: I believe `QPalette` is in `QtGui`?
```suggestion
            focus_opt.backgroundColor = option.palette.color(
                (
                    QtGui.QPalette.Normal
                    if option.state & QtWidgets.QStyle.State_Enabled
                    else QtGui.QPalette.Disabled
                ),
                (
                    QtGui.QPalette.Highlight
                    if option.state & QtWidgets.QStyle.State_Selected
                    else QtGui.QPalette.Window
                )
            )
```

Comment (BigRoy) [reply client/ayon_core/tools/utils/delegates.py]: Fixed with https://github.com/ynput/ayon-core/pull/593/commits/4264af63aff45a03e48b2ca6f8868747a24039bc

**Comment by BigRoy** [comment]
Nice.

All works.

I do wonder if the combobox (main entry, not the list) needs more padding/margins:

image

**Comment by BigRoy** [comment]
By the way, is the label "community contribution" correct on this PR?

**Comment by iLLiCiTiT** [comment]
> I do wonder if the combobox (main entry, not the list) needs more padding/margins:

Where did you take the screenshot?

image


> By the way, is the label "community contribution" correct on this PR?

Yes, I do it in my free time.

**Comment by BigRoy** [comment]
> Where did you take the screenshot?

Good question! :)

This is in Maya 2024:

image

This is in Houdini 20.0.703:

image

This is in Fusion 18:

image

**Review by m-u-r-p-h-y — APPROVED** [review]
Everything works fine. . .

image

**Pull request merged**